### PR TITLE
CBG-1004: Avoid rendering attachments in browser

### PR DESCRIPTION
**Pull request by JRascagneres**
A number of content-types will have a content-disposition header added to it so when the attachment URL is visited the file is instead downloaded rather than displayed in the browser. This can be bypassed with a config option.

Content-Disposition was decided on as it is already used on the admin port, and to avoid any potential issues with changing content-type: 
Issues: https://github.com/couchbase/sync_gateway/issues/720
Commit: https://github.com/couchbase/sync_gateway/commit/266cf96de8b1be5d2e11bf3b87dcfa8d9bc9321c

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition

### CODE REVIEW

File: rest/api_test.go [hunk 1]
```diff
@@ -5106,3 +5106,76 @@ func TestImportOnWriteMigration(t *testing.T) {
...
+		putContentType                string
+		expectedContentDispositionSet bool
+	}
+
+	tests := []attTest{
+		{
+			setContentType:                true,
+			putContentType:                "text/html",
+			expectedContentDispositionSet: true,
+		},
```
Comment (bbrks) [inline rest/api_test.go:5125]: Might be worth adding a test case with a parameter

```suggestion
		{
			setContentType:                true,
			putContentType:                "text/html",
			expectedContentDispositionSet: true,
		},
		{
			setContentType:                true,
			putContentType:                "text/html; charset=utf-8",
			expectedContentDispositionSet: true,
		},
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Maybe a test case to cover mime types with a parameter. Other than that, it looks good to me.
We should run this new Content-Disposition approach (and list of "unsafe" mimes) past @adamcfraser to make sure he's OK with it.

File: rest/doc_api.go [hunk 2]
```diff
@@ -209,10 +209,37 @@ func (h *handler) handleGetAttachment() error {
...
+	// unless overridden by config option. CBG-1004
+	contentType, contentTypeSet := meta["content_type"].(string)
+	if contentTypeSet {
 		h.setHeader("Content-Type", contentType)
 	}
+
+	if !h.db.ServeInsecureAttachmentTypes {
+
+		if contentTypeSet {
+			contentTypeFirst := strings.Split(contentType, ";")[0]
```
Comment (bbrks) [inline rest/doc_api.go:232]: As discussed earlier, add comment to explain this.

**Review by bbrks — CHANGES_REQUESTED** [review]
Looks good apart from the comment we talked about earlier explaining the optional mime-type parameters.

File: rest/doc_api.go (hunk 2 above)
Comment (JRascagneres) [reply rest/doc_api.go:232]: Ah yeah woops

**Review by bbrks — APPROVED** [review]

**Pull request merged**